src/commands/admin/prune.ts

```diff
@@ -0,0 +1,105 @@
+			.setFooter(`This command will expire in ${PRUNE_TIMEOUT} seconds`);
+
+		const confirmBtns = [
+			new MessageButton({ label: 'Cancel', customId: 'cancel', style: 'SECONDARY' }),
+			new MessageButton({ label: 'Proceed', customId: 'proceed', style: 'DANGER' })
+		];
+
+		const confirmMsg = await msg.channel.send({
+			embeds: [confirmEmbed],
+			components: [new MessageActionRow({ components: confirmBtns })]
+		});
+
+		const collector = msg.channel.createMessageComponentCollector({
+			filter: i => i.message.id === confirmMsg.id, time: PRUNE_TIMEOUT * 1000
+		});
+
+		const countdown = setInterval(() => this.countdown(confirmMsg, --timeout, confirmBtns, confirmEmbed), 1000);
+
+		collector.on('collect', async interaction => {
+			if (!interaction.isButton()) return;
+			if (interaction.user.id !== msg.author.id) {
+				await interaction.reply({
+					content: 'You cannot respond to a command you did not execute',
+					ephemeral: true
+				});
+				return;
+			}
+			interaction.deferReply({ ephemeral: true });
+			clearInterval(countdown);
+
+			confirmBtns.forEach(btn => btn.setDisabled(true));
+
+
+			if (interaction.customId === 'cancel') {
+				confirmEmbed.setColor('BLUE')
+					.setTitle(`Prune cancelled, ${msg.member.displayName} got cold feet!`);
+				confirmMsg.edit({ embeds: [confirmEmbed], components: [new MessageActionRow({ components: confirmBtns })] });
+			} else {
+				confirmEmbed.setTitle(`<a:loading:928003042954059888> Pruning ${toKick.size} members...`);
+				confirmMsg.edit({ embeds: [confirmEmbed], components: [new MessageActionRow({ components: confirmBtns })] });
+
+				const awaitedKicks: Promise<GuildMember>[] = [];
+				toKick.forEach(member => {
+					awaitedKicks.push(member.kick('Pruned by the prune command'));
+					console.log(`pruning ${member.displayName}`);
+					return;
+				});
+				await Promise.all(awaitedKicks);
+
+				confirmEmbed.setTitle(`:white_check_mark: Pruned ${toKick.size} members!`);
+				confirmMsg.edit({ embeds: [confirmEmbed], components: [new MessageActionRow({ components: confirmBtns })] });
+			}
+			collector.stop();
+		}).on('end', async collected => {
+			const validCollected = collected.filter(i => i.isButton()
+			&& i.message.id === confirmMsg.id
+			&& i.user.id === msg.author.id);
+
+			if (validCollected.size === 0) {
+				clearInterval(countdown);
+				confirmBtns.forEach(btn => btn.setDisabled(true));
+				confirmEmbed.setColor('BLUE').setDescription('Prune timed out.');
+			}
+			confirmEmbed.setFooter('');
+			confirmMsg.edit({ embeds: [confirmEmbed], components: [new MessageActionRow({ components: confirmBtns })] });
+
+			collected.forEach(interaction => {
+				if (validCollected.has(interaction.id)) interaction.followUp({ content: 'Done!' });
+			});
+		});
+
+		return;
+	}
+
+	countdown(msg: Message, timeout: number, confirmBtns: MessageButton[], confirmEmbed: MessageEmbed): void {
+		confirmEmbed.setFooter(`This command will expire in ${timeout} seconds`);
```

Comment:
This could read `This command will expire in 1 seconds`.

Reply:
:(